### ISSUE TITLE
ServiceLocatorAwareInterface is deprecated

### ISSUE THREAD

**Issue by sharksoft** [SocalNick/ScnSocialAuth#236]
Deprecated: ServiceManagerAwareInterface is deprecated and will be removed in version 3.0, along with the ServiceManagerAwareInitializer. Please update your class ScnSocialAuth\Authentication\Adapter\HybridAuth to remove the implementation, and start injecting your dependencies via factory instead

**Comment by SocalNick** [comment]
@sharksoft would you consider sending a PR to fix this?

**Comment by sharksoft** [comment]
Hi @SocalNick, in zend3 (and latest version of zf2) the use of factory is mandatory. I'm updating my projects to ZF3, maybe I can help you (not now) with a fork

**Comment by shinesoftware** [comment]
Hi @SocalNick there is a problem with the "PR". Could you, please check it? 
thanks

**Comment by SocalNick** [comment]
@shinesoftware I think you'll need to fix the conflicts in #238

**Comment by shinesoftware** [comment]
@SocalNick I have tried to resolve by "Resolve Conflict" but Github crashes
![screenshot1481878011 1x](https://cloud.githubusercontent.com/assets/1553035/21256806/c7daa498-c374-11e6-80ad-cffd15c545bd.png)
!

**Comment by SocalNick** [comment]
@shinesoftware I started to fix the conflicts in #238 but there seemed to be some unnecessary changes. Could you take a look at my alternative PR #239 to see if I missed anything?

**Comment by shinesoftware** [comment]
I have checked your PR and seems good! maybe my commit contains the old controller changes already included in my https://github.com/SocalNick/ScnSocialAuth/pull/235 and github doesn't merge the new files. @SocalNick

**Comment by SocalNick** [comment]
Oh , that could be. PRO TIP: Don't commit to master of your fork, always commit to a branch and submit PRs from your branch to the upstream master branch. Once your PR is merged, simply delete your branch and pull upstream master into master on your fork.
